### PR TITLE
doc: render math with HTML to make it selectable (fix #8453)

### LINKED ISSUE #8453: rtd: text on images is unhelpful
I reach to [Test Suite Fields docs](https://cabal.readthedocs.io/en/latest/buildinfo-fields-reference.html#test-suite-fields) and see:

I'd really love to copy and paste the exitcode stuff but I can't because it's a picture. Generally this page seems to overuse images. I propose we put copy-pasteable bits (words) in plain text, if possible.

### PR DESCRIPTION
**Pull request by ulysses4ever**
**Template Β: This PR does not modify `cabal` behaviour (documentation, tests, refactoring, etc.)**

Include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).

### CODE REVIEW

**Comment by geekosaur** [comment]
It doesn't seem like the text in the field reference of the RTD build is selectable yet. Am I checking the right thing?

**Comment by ulysses4ever** [comment]
So far, it just looks awful. I blame MathJax 3. Will probably try to roll back to 2.

**Comment by ulysses4ever** [comment]
It appears scalable though. https://cabal--9361.org.readthedocs.build/en/9361/buildinfo-fields-reference.html

**Comment by geekosaur** [comment]
So I was looking at the right thing, at least.

**Comment by kukimik** [comment]
If you wish to bump the version of MathJax 2 along the way, take note of the following bug (described [in the docs](https://docs.mathjax.org/en/latest/upgrading/v2.html)):

> There is, however, a bug in latest.js in versions 2.7.5 and below; when the current version is 3.0 or higher, latest.js will not use the highest version of 2.x, but instead will use the version from which latest.js has been taken. For example, if you load latest.js from version 2.7.3, it currently is giving you version 2.7.5 as the latest version, when version 3 is released to the CDN, your pages will revert to using version 2.7.3 again. This behavior has been corrected in version 2.7.6, so if you change to loading latest.js from version 2.7.6, you should get the latest 2.x version regardless of the presence of version 3 on the CDN.

So using `2.7.5/latest.js` won't give you the latest 2.x version (but `2.7.6/latest.js` or `2.7.7/latest.js` should).

I guess `TeX-AMS_CHTML` should be the right [config](https://docs.mathjax.org/en/v2.7-latest/config-files.html). Didn't try if it works.

**Comment by geekosaur** [comment]
Does that need to specify 2.5.6 or 2.5.7 per @kukimik's comment?

**Comment by geekosaur** [comment]
And it looks like the text is indeed selectable with MathJax 2.

**Comment by ulysses4ever** [comment]
I think it was selectable even with MathJax 3 but it looked a bit worse, I felt. So, I rolled back to 2. Please, check it out. (the link is the same: https://cabal--9361.org.readthedocs.build/en/9361/buildinfo-fields-reference.html)

Converting from Draft to Ready.

**Comment by geekosaur** [comment]
I was unable to select anything with MathJax 3, and yes, it looked hideous in Chrome Linux.

**Review by geekosaur — APPROVED** [review]

**Comment by ulysses4ever** [comment]
> I was unable to select anything with MathJax 3, and yes, it looked hideous in Chrome Linux.

Hm, FTR I'm using Firefox...

**Comment by kukimik** [comment]
LGTM on Linux Firefox and Chromium. The only issue I noticed is a line that is too long, but it happens with SVG too:

image

**Comment by ulysses4ever** [comment]
@kukimik thanks for checking! Indeed, the overflow doesn't look nice but (a) I don't see an easy way to fix it, (b) this is unrelated to this PR. So, I suggest we leave for future work...

File: doc/conf.py [hunk 1]
```diff
@@ -102,8 +102,8 @@
 # Output file base name for HTML help builder.
 htmlhelp_basename = 'CabalUsersGuide'
 
-# MathJax to use SVG rendering by default
-mathjax_path = 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_SVG'
+# MathJax to use HTML rendering by default (makes the text selectableable, see #8453)
```
Comment (chreekat) [inline doc/conf.py:105]:
```suggestion
# MathJax to use HTML rendering by default (makes the text selectable, see #8453)
```

**Comment by ffaf1** [comment]
For people who would like to compare:

1) https://cabal--9361.org.readthedocs.build/en/9361/buildinfo-fields-reference.html
2) https://cabal.readthedocs.io/en/stable/buildinfo-fields-reference.html

File: doc/conf.py (hunk 1 above)
Comment (ulysses4ever) [reply doc/conf.py:105]: Oh, thank you!How hard could it bee to not slip in one sentence...

**Pull request merged**